packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -0,0 +1,103 @@
+type Value<T, K extends string> = K extends keyof T ? T[K] : unknown;
+
+export type NatifiedArray<T> = T[] & NativeArray<T>;
+
+interface EmberArray<T> {
+  length: number;
+  objectAt(idx: number): T | undefined;
+}
+declare class EmberArray<T> {
+  objectsAt(indexes: number[]): Array<T | undefined>;
+  get firstObject(): T | undefined;
+  get lastObject(): T | undefined;
+  slice(beginIndex?: number, endIndex?: number): NatifiedArray<T>;
+  indexOf(object: T, startAt?: number): number;
+  lastIndexOf(object: T, startAt?: number): number;
+  forEach<Target>(
+    callback: (this: Target, item: T, index: number, arr: this) => void,
+    target?: Target
+  ): this;
+  getEach<K extends string>(key: K): NatifiedArray<Value<T, K>>;
+  setEach<K extends string>(key: K, value: Value<T, K>): this;
+  map<R, Target>(
+    callback: (this: Target, item: T, index: number, arr: this) => R,
+    target?: Target
+  ): NatifiedArray<R>;
+  mapBy<K extends string>(key: K): NatifiedArray<Value<T, K>>;
+  filter<Target>(
+    callback: (this: Target, item: T, index: number, arr: this) => boolean,
+    target?: Target
+  ): NatifiedArray<T>;
+  reject<Target>(
+    callback: (this: Target, item: T, index: number, arr: this) => boolean,
+    target?: Target
+  ): NatifiedArray<T>;
+  filterBy(key: string): NatifiedArray<T>;
+  rejectBy(key: string): NatifiedArray<T>;
+  find<Target = null>(
```

Comment:
We should make this match the types from `Array`, so `Target` should default to `void` rather than `null`.

```suggestion
  find<Target = void>(
```